waspc/cli/src/Wasp/Cli/Command/CreateNewProject.hs

```diff
@@ -45,81 +60,36 @@ parseProjectInfo name
   where
     appName = kebabToCamelCase name
 
-createNewProject :: String -> Command ()
-createNewProject name =
-  case parseProjectInfo name of
-    Right projectInfo -> createNewProject' projectInfo
-    Left parsedError ->
-      throwError $
-        CommandError "Project creation failed" parsedError
-
-createNewProject' :: ProjectInfo -> Command ()
-createNewProject' (ProjectInfo projectName appName) = do
+createWaspProjectDir :: ProjectInfo -> Command ()
+createWaspProjectDir projectInfo = do
+  absWaspProjectDir <- getAbsoluteWaspProjectDir projectInfo
+  dirExists <- doesDirExist $ toPathAbsDir absWaspProjectDir
+  if dirExists
+    then throwProjectCreationError $ show absWaspProjectDir ++ " is an existing directory"
+    else liftIO $ do
+      initializeProjectFromSkeleton absWaspProjectDir
+      writeMainWaspFile absWaspProjectDir projectInfo
+
+getAbsoluteWaspProjectDir :: ProjectInfo -> Command (Path System Abs (Dir WaspProjectDir))
```

Comment:
Not quite:
- `getAbsoluteWaspProjectDir`  is used to create up with an absolute path for a Wasp project you wish to create in the current directory.
- `findWaspProjectRootDirFromCwd` is used to find the root directory inside a Wasp project (this is the thing that uses the infamous `.wasproot`).

If I called the latter function, it would throw an error saying "Couldn't find project root, are you sure you're inside the Wasp project?"

Reply:
Ah yeah, good point, makes sense 👍🏻